system/lib/pthread/proxying.c

```diff
@@ -138,12 +191,51 @@ em_proxying_queue* emscripten_proxy_get_system_queue(void) {
   return &system_proxying_queue;
 }
 
+// The head of the zombie list. Its mutex protects access to the list and its
+// other fields are not used..
+static em_proxying_queue zombie_list_head = {.prev = &zombie_list_head,
+                                             .next = &zombie_list_head,
+                                             .mutex =
+                                               PTHREAD_MUTEX_INITIALIZER};
```

Comment:
Why isn't the zombie list empty to begin with? IIUC this code sets it to contain one proxying queue.

Reply:
You can think of the `zombie_list_head` as the sentinel value we use to mark the beginning and end of the list and not as  a normal element in the list. With that understanding, the list does start out empty. The reason to avoid using NULL as the sentinel is that we would then have to check for null in the code to insert or remove list elements, complicating and slowing down the code, which is otherwise branch-free.

Reply:
I see, thanks. Makes sense, I guess the logic to special-case the head is smaller than the other costs.